core/schema/container.go

```diff
@@ -1011,7 +1011,9 @@ type containerWithDefaultArgs struct {
 }
 
 func (s *containerSchema) withDefaultArgs(ctx context.Context, parent *core.Container, args containerWithDefaultArgs) (*core.Container, error) {
-	return parent.UpdateImageConfig(ctx, func(cfg specs.ImageConfig) specs.ImageConfig {
+	c := parent.Clone()
+	c.DefaultArgs = args.Args
```

Comment:
Ok, I see.

Because `DefaultArgs` is only ever set to the same as `Cmd`, probably doesn't make sense to track it as a slice. Maybe we can just track `DefaultArgs bool` for explicitly tracking this error.